src/auth/googleauth.ts

```diff
@@ -603,6 +603,15 @@ export class GoogleAuth {
       return this.cachedCredential.projectId;
     }
 
+    // Ensure the projectId is loaded from the keyFile if available.
+    if (this.keyFilename) {
+      const creds = await this.getClient();
+
+      if (creds && creds.projectId) {
+        return creds.projectId;
```

Comment:
@JustinBeckwith made a small amendment to your commit, let me know if you hate it!